### PR TITLE
Bbinit implicit cpl

### PR DESCRIPTION
**Pull request by atotoun**

### CODE REVIEW

File: src/m2n/M2N.cpp [hunk 1]
```diff
@@ -122,6 +124,7 @@ void M2N::acceptSlavesPreConnection(
   for (const auto &pair : _distComs) {
     pair.second->acceptPreConnection(acceptorName, requesterName);
     _areSlavesConnected = _areSlavesConnected && pair.second->isConnected();
+    std::this_thread::sleep_for(std::chrono::seconds(10));
```
Comment (uekerman) [inline src/m2n/M2N.cpp]:
```suggestion
   // @todo Remove workaround as soon as we have a proper solution
    std::this_thread::sleep_for(std::chrono::seconds(10));
```

File: src/m2n/M2N.cpp [hunk 2]
```diff
@@ -5,6 +5,8 @@
 #include "mesh/Mesh.hpp"
 #include "utils/Event.hpp"
 #include "utils/MasterSlave.hpp"
+#include<thread>
```
Comment (uekerman) [inline src/m2n/M2N.cpp]:
```suggestion
// @todo Remove workaround as soon as we have a proper solution
#include<thread>
```

File: src/m2n/M2N.cpp [hunk 3]
```diff
@@ -135,6 +138,7 @@ void M2N::requestSlavesPreConnection(
   for (const auto &pair : _distComs) {
     pair.second->requestPreConnection(acceptorName, requesterName);
     _areSlavesConnected = _areSlavesConnected && pair.second->isConnected();
+    std::this_thread::sleep_for(std::chrono::seconds(10));
```
Comment (uekerman) [inline src/m2n/M2N.cpp]:
```suggestion
    // @todo Remove workaround as soon as we have a proper solution
    std::this_thread::sleep_for(std::chrono::seconds(10));
```

File: src/m2n/PointToPointCommunication.cpp [hunk 4]
```diff
@@ -540,14 +540,18 @@ void PointToPointCommunication::requestPreConnection(std::string const &acceptor
   requests.reserve(localConnectedRanks.size());
   _connectionDataVector.reserve(localConnectedRanks.size());
 
-  std::set<int> acceptingRanks(localConnectedRanks.begin(), localConnectedRanks.end());
+  std::set<int> acceptingRanks;
+  for (auto &i : localConnectedRanks)
+    acceptingRanks.emplace(i);
+
+  // std::set<int> acceptingRanks(localConnectedRanks.begin(), localConnectedRanks.end());
```
Comment (uekerman) [inline src/m2n/PointToPointCommunication.cpp]:
```suggestion
```

File: src/partition/ProvidedBoundingBox.cpp [hunk 5]
```diff
@@ -208,8 +217,8 @@ void ProvidedBoundingBox::communicate()
   // each rank sends its mesh partition to connected remote ranks
   _m2ns[0]->broadcastSendLocalMesh(*_mesh);
   
-  createOwnerInformation();  
-  computeVertexOffsets();
+  createOwnerInformation();    
+  // computeVertexOffsetsBB(_mesh->vertices().size());
```
Comment (uekerman) [inline src/partition/ProvidedBoundingBox.cpp]: Why don't you use this function here?

File: src/partition/ProvidedBoundingBox.cpp [hunk 6]
```diff
@@ -148,11 +148,14 @@ void ProvidedBoundingBox::communicate()
 
     // set global indexes for master rank mesh partition 
     for(int i=0; i<(int)_mesh->vertices().size(); i++){
-      _mesh->getVertexDistribution()[0].push_back(vertexCounter);
+      // _mesh->getVertexDistribution()[0].push_back(vertexCounter);
```
Comment (uekerman) [inline src/partition/ProvidedBoundingBox.cpp]:
```suggestion
```

File: src/partition/ProvidedBoundingBox.cpp (hunk 5 above)
Comment (atotoun) [reply src/partition/ProvidedBoundingBox.cpp]: I have deactivated it to test! Forgot to re-activate! 
Thanks for reminding!

**Pull request merged**